contrib/database/sql/sql.go

```diff
@@ -236,3 +232,29 @@ func Open(driverName, dataSourceName string, opts ...Option) (*sql.DB, error) {
 	}
 	return OpenDB(&dsnConnector{dsn: dataSourceName, driver: d}, opts...), nil
 }
+
+func processOptions(cfg *config, driverName string, opts ...Option) {
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	checkDBMPropagation(cfg, driverName)
+}
+
+func checkDBMPropagation(cfg *config, driverName string) {
+	fullModeSupported := func() bool {
+		unsupportedDrivers := []string{"sqlserver", "oracle"}
+		for _, dr := range unsupportedDrivers {
+			if dr == driverName {
+				return false
+			}
+		}
+		return true
+	}
+	if cfg.dbmPropagationMode == tracer.DBMPropagationModeFull && !fullModeSupported() {
+		log.Warn("Using DBM_PROPAGATION_MODE in 'full' mode is not supported for %s. See "+
```

Comment:
I think it might make sense here to state that we're downgrading to `service` mode.